packages/twenty-front/src/modules/apollo/optimistic-effect/utils/triggerUpdateRecordOptimisticEffectByBatch.ts

```diff
@@ -105,7 +103,10 @@ export const triggerUpdateRecordOptimisticEffectByBatch = ({
           }
 
           if (updatedRecordShouldBeRemovedFromRootQueryEdges) {
-            rootQueryNextEdges.splice(updatedRecordIndexInRootQueryEdges, 1);
+            rootQueryNextEdges = rootQueryNextEdges.filter(
+              (cachedEdge) =>
```

Comment:
splicing was causing an issue here because it modified the indexes of elements in rootQueryNextEdges, which was problematic when there were multiple iterations